### ISSUE TITLE
allure-mocha:  Cannot run example from docs since allure-runtime gives an undefined property

### ISSUE THREAD

**Issue by jamesmortensen** [allure-framework/allure-js#509]
**Describe the bug**

When attempting to use the simple example from the readme, we discovered that the `allure` variable is undefined, so we cannot call any of the "extra information" methods, such as `steps` and `attachments`.

**To Reproduce**

1.  Clone the following small repro example:  

```
$ git clone https://github.com/jamesmortensen/allure-js-runtime-issue.git
```

2.  Run the tests with `npm test`
3.  Observe the error that `allure` is undefined.

**Expected behavior**

According to the documentation, if we want to add extra information to the report, we can import allure/runtime and then use methods, such as allure.epic, story, description, etc.  We expect the tests to complete, with one passing and one failing and for the extra information to appear in the report.

**Desktop (please complete the following information):**
 - OS: macOS 12.4 ARM64
 - Browser N/A
 - Version N/A

We are running Node.js v16.17 and allure-mocha 2.0.0-beta.19

**Additional context**

See the [Usage](https://github.com/allure-framework/allure-js/tree/master/packages/allure-mocha#usage) section of the docs to see the instructions.

**Stacktrace:**

```
TypeError: Cannot read properties of undefined (reading 'epic')
      at Context.<anonymous> (test/example.spec.js:21:16)
      at processImmediate (node:internal/timers:466:21)
      at process.callbackTrampoline (node:internal/async_hooks:130:17)
```

**Comment by epszaw** [comment]
Thank you for the issue. It's strongly relates to #486 and depends on mochajs/mocha#4922

**Comment by jamesmortensen** [comment]
Actually, my colleague dug into this pretty heavily today, and she got it working!  

We discovered two causes.  First, she discovered that the way the docs said to require the runtime was incorrect.  Instead of:

```
const { allure } = require("allure-mocha/runtime");
```
She did this:
```
const runtime = require("allure-mocha/runtime");
```
Then we could do `runtime.allure.epic('This is epic!');`

We also discovered that it is because Mocha, or perhaps allure, wasn't able to handle both the default spec reporter and the allure-mocha reporter together.  If we removed the spec reporter, then the runtime is defined. But with both, runtime.allure was undefined.

We used the mocha-multi-reporter module to configure both reporters together.  If you need more details on this please let me know.  I do think it is worth updating the docs with this information, as it may help others.

**Comment by epszaw** [comment]
It depends on version of the packages.
The latest one doesn't work properly because at the same moment we can have multiple tests (in parallel mode) and any label assignment by the runtime can add label to a wrong test or suite.

**Comment by jamesmortensen** [comment]
@epszaw if it helps, we are on the latest everything:

- Mocha 10.1.0
- allure-mocha: 2.0.0-beta.19
- Node 16.17

We do run tests in parallel, but we use a variation of a script shared by an allure collaborator:  https://github.com/allure-framework/allure-js/issues/245#issuecomment-1176254785.  The script splits up all of the spec files and runs them as separate, independent mocha processes. So we don't run into any of these issues when running in parallel.

**Comment by delatrie** [comment]
The original issue has been fixed for a lengthy time since 2.5.0 (see #730).
Now, we're transitioning to 3.0, which exposes the API via allure-js-commons. The examples [here](https://github.com/allure-framework/allure-js/tree/main/packages/allure-mocha#usage) are updated and tested.

The new version is used like this:

```js
import { it } from "mocha";
import { label } from "allure-js-commons";

it("bar", async () => {
  await label("foo", "bar");
});
```

The legacy version is also available (note the absence of the `.js` extension in `"allure-mocha/runtime"`):

```js
import { it } from "mocha";
import { allure } from "allure-mocha/runtime";

it("bar", async () => {
  await allure.label("foo", "bar");
});
```
